### ISSUE TITLE
Provider `pydict` is flaky 

### ISSUE THREAD

**Issue by bverhoeve** [joke2k/faker#2045]
* Faker version: 25.0.1
* OS: macOS Sonoma

Brief summary of the issue goes here.
When using the `pydict` provider, this sometimes results in a `ValueError` in the underlying `random_elements` provider:
```
ValueError: Sample length cannot be longer than the number of unique elements to pick from.
```
This error does not pop up all the time but it seems to occur for some combination. Chances of it happening seem to increase with a higher value for the `nb_elements` parameter

### Steps to reproduce

1. Run `faker.pydict(nb_elements=1000)` a couple of times. After a while the error pops up

### Expected behavior
A random `dict` is generated

### Actual behavior
A `ValueError` is raised.

**Comment by mileswatsonbjss** [comment]
We have a potential fix, will post the PR here when it is ready for review

**Comment by bverhoeve** [comment]
Great news! What was the issue?

**Comment by mileswatsonbjss** [comment]
Here is the PR, If you could both please review when you have a chance (https://github.com/joke2k/faker/pull/2067)

Details re. what caused the issue etc.. are all in the PR.

**Comment by fcurella** [comment]
Released in v26.0.0

**Comment by bverhoeve** [comment]
Awesome, thanks guys!
